doc/decisions/library_directory_structure.md

```diff
@@ -2,21 +2,75 @@
 
 ## Problem
 
-Currently src/libs/elektra contains the source of several libraries. Each of the libraries should have its own folder.
+Currently `src/libs/elektra` contains the source of several libraries. Each of the libraries should have its own folder.
 Furthermore, we need to allow rust source to exist in parallel.
 
 ## Constraints
 
+- It must be possible to set up include-paths via CMake so that `#include "header.h"` works the same in the build directory and when installed.
+
 ## Assumptions
 
 ## Considered Alternatives
 
 ## Decision
 
+We use this directory tree (YAML syntax for highlighting):
+
+```yaml
+doc:
+examples:
+scripts:
+src:
+  include:
+   # headers that don't belong to a specific library
+  libs:
+    core-c: C implementation of libelektra-core.so
+    core-rust: Rust implementation of libelektra-core.so
+    opts: libelektra-opts.so has only a single implementation, so no suffix
+    mount-c: mounting library written in C
+  # plugins and tools use the same suffix idea as above
+  plugins:
+  tools:
+  bindings:
+    rust: Rust bindings
+    cpp: C++ bindings
+    java: Java bindings
+  tests:
+    shell: shell script tests
+    cframework: framework for C tests
+    gtest-framework: framework for C++ tests
+    abi: ABI compatibility tests
+  # build tool configuration files to create a single root project for every language
+  # this is improves IDE support
+  CMakeLists.txt
+  build.gradle
+  Cargo.toml
+```
+
+Unit tests (\*) (e.g. the old `tests/ctest`, `tests/kdb`) should be bundled with the code they are testing:
+e.g. in Rust test code is in the same file, in C/C++ it should be in the same folder and in Java a separate source set in the same module is used.
+This makes it easier to find the tests for a library, and also creates a uniform structure (in Rust or Java it would be harder to put tests into the current structure).
+System tests that test multiple components live in the `src/tests` folder.
```

Comment:
It may be worded badly, but I don't think it's a big task. It's mostly about moving stuff from `tests/ctest` into the appropriate `src/libs/[lib]` folder. Then rest just needs to be moved from `tests` to `src/tests`. For Java/Rust the tests are already set up as described here.